### PR TITLE
TM-965 create republish command

### PR DESCRIPTION
**Pull request by kallosz**

### CODE REVIEW

File: src/SevenTag/Api/AppBundle/Command/RepublishContainerCommand.php [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+     * @return int|null|void
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $containerId = $input->getOption('containerId');
+
+        if ($containerId) {
+            $containers = $this->getSingleContainerToProcess($containerId);
+        } else {
+            $containers = $this->getAllPublishedContainersToProcess();
```
Comment (zientalak) [inline src/SevenTag/Api/AppBundle/Command/RepublishContainerCommand.php]: You don't need else statement. Move to private method with opimistic condition or to ternary operator, eq.:

```
private function getContainersToProcess($containerId)
{
	if($containerId) {
		return $this->getSingleContainerToProcess($containerId);
	}

	return $this->getAllPublishedContainersToProcess();
}
```

File: src/SevenTag/Api/AppBundle/Command/RepublishContainerCommand.php [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        $progress->finish();
+
+        $output->writeln('');
+        $output->writeln('<info>Republish completed.</info>');
+    }
+
+    /**
+     * @param $containerId
+     * @return ArrayCollection|null
+     */
```
Comment (zientalak) [inline src/SevenTag/Api/AppBundle/Command/RepublishContainerCommand.php]: You don't need return nul in this placel. You can return ArrayCollection with empty collection data and then instead checking null in line 80 just make sure that collection is not empty:

```
if($containers->isEmpty() {}
```

File: src/SevenTag/Api/AppBundle/Command/RepublishContainerCommand.php [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+use SevenTag\Api\ContainerBundle\Entity\ContainerRepository;
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Helper\DialogHelper;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+class RepublishContainerCommand extends ContainerAwareCommand
```
Comment (zientalak) [inline src/SevenTag/Api/AppBundle/Command/RepublishContainerCommand.php]: If you have registered command as a service, you don't need ContainerAwareCommand. Use Command instead.

File: src/SevenTag/Api/AppBundle/Command/RepublishContainerCommand.php [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        /** @var DialogHelper $dialog */
+        $dialog = $this->getHelper('dialog');
+        if (!$dialog->askConfirmation($output, sprintf('<question>%d containers will be republished. Are you sure?', $containers->count(), false))) {
+            return;
+        }
+
+        $progress = new ProgressBar($output, $containers->count());
+
+        foreach ($containers as $container) {
+            $this->eventDispatcher->dispatch(Events::CONTAINER_REPUBLISHED, new ContainerLibraryEvent($container));
```
Comment (zientalak) [inline src/SevenTag/Api/AppBundle/Command/RepublishContainerCommand.php]: Overengineering. We don't need different event constans for republish. Container might be published in our application several times and we call it publish/republish but it means the same operation from business point of view.

Comment (zientalak) [inline src/SevenTag/Api/AppBundle/Command/RepublishContainerCommand.php]: This code does not solve issue. In controller during publish we have code:

```
        $versionManager = $this->get('seven_tag_app.version_manager');
        $container = $versionManager->publish($container);

        $this->get('event_dispatcher')
            ->dispatch(Events::CONTAINER_PUBLISHED, new ContainerLibraryEvent($container));
```

You have forgotten use version manager to publish container. Event only notify listeners about publish operation (in order to save tag tree and container in filesystem or/and CDN).

File: src/SevenTag/Api/AppBundle/Command/RepublishContainerCommand.php (hunk 2 above)
Comment (kallosz) [inline src/SevenTag/Api/AppBundle/Command/RepublishContainerCommand.php]: ArrayCollection is not always returned.

File: src/SevenTag/Api/ContainerBundle/Service/RepublishContainer.php [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+        if (null === $containers) {
+            $containers = [];
+        }
+
+        return new ArrayCollection($containers);
+    }
+
+    /**
+     * @return \SevenTag\Api\ContainerBundle\Entity\ContainerRepository
```
Comment (michalsikora) [inline src/SevenTag/Api/ContainerBundle/Service/RepublishContainer.php]: simplify class

**Pull request merged**